pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -100,22 +100,18 @@ func (mixerplugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.Mu
 		"context.reporter.kind": attrStringValue("outbound"),
 	}
 
-	switch in.ListenerProtocol {
-	case plugin.ListenerProtocolHTTP:
-		filter := buildOutboundHTTPFilter(in.Env.Mesh, attrs, in.Node)
-		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, filter)
-		}
-		return nil
-	case plugin.ListenerProtocolTCP:
-		filter := buildOutboundTCPFilter(in.Env.Mesh, attrs, in.Node, in.Service, in.Push)
-		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].TCP = append(mutable.FilterChains[cnum].TCP, filter)
+	httpFilter := buildOutboundHTTPFilter(in.Env.Mesh, attrs, in.Node)
+	tcpFilter := buildOutboundTCPFilter(in.Env.Mesh, attrs, in.Node, in.Service, in.Push)
+
+	for cnum, lFilterChain := range mutable.Listener.FilterChains {
```

Comment:
what's the difference between mutable.Listener.FilterChains and mutable.FilterChains? Why is it looping over one and modifying the other?

Reply:
I have the same doubt, it's hard to understand.